forc-pkg/src/source/git/mod.rs

```diff
@@ -310,10 +327,24 @@ impl From<Pinned> for source::Pinned {
     }
 }
 
+impl FromStr for GitUrl {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        if s.starts_with("git") {
+            // TODO: We should probably parse for and detect any errors with this as well.
+            Ok(Self::Ssh(s.to_string()))
```

Comment:
Ahh, we might want to be careful about our variant names for this `GitUrl` type.

E.g. I think normal git also supports ssh URLs like `ssh://`, however it looks like these would currently end up in the `Http` variant which would be a little unintuitive :sweat_smile:

Git's URL handling is pretty diverse - here's their URL normalization testing: https://github.com/git/git/blob/77bd3ea9f54f1584147b594abc04c26ca516d987/t/t0110-urlmatch-normalization.sh

Perhaps we could use the [gitoxide URL representation](https://docs.rs/gix-url/0.16.0/gix_url/struct.Url.html) for validation rather than make our own `GitUrl` type?

Reply:
Ahh that makes sense, I will look into it 👍